File: app-gui/src/main/java/markov/Application.java
package markov;

import markov.input.Data;
import markov.input.DataLoader;
import markov.manual.Manual;
import markov.manual.ManualController;
import markov.map.MapReader;
import markov.map.SimulationMap;
import markov.map.SimulationMapController;
import markov.movement.Movement;
import markov.movement.MovementController;
import markov.movement.MovementReader;
import markov.random.OneToHundredGenerator;
import markov.simulation.SimulationController;
import markov.simulation.SimulationTime;

import javax.swing.*;
import java.awt.*;

public class Application {
    public static void main(String[] args) {
        MessageLogger messageLogger = new MessageLogger();
        SimulationMap map = new SimulationMapController(messageLogger).readMap(
                MapReader.INSTANCE.read(DataLoader.INSTANCE.load(Data.MAP))
        );
        Movement movement = new MovementController(
                map.getSize(),
                messageLogger
        ).readMovement(MovementReader.INSTANCE.read(DataLoader.INSTANCE.load(Data.PROBABILITY)));
        SimulationPanel simulationPanel = new SimulationPanel(movement);
        MainFrame mainFrame = new MainFrame(messageLogger, simulationPanel);
        Manual manual = new ManualController(simulationPanel).findBestManual(map, movement);
        new SimulationController(simulationPanel).startFrom(
                map,
                new SimulationTime(10), movement, OneToHundredGenerator.INSTANCE
        );
        simulationPanel.paintSimulation();
    }
}

File: app-cli/src/main/kotlin/cli/Application.kt
package cli

import markov.input.Data
import markov.input.DataLoader
import markov.cost.CostMapController
import markov.map.MapReader
import markov.map.SimulationMapController
import markov.movement.MovementController
import markov.movement.MovementReader
import markov.output.Message
import markov.output.MessageType
import markov.random.OneToHundredRandomGenerator
import markov.simulation.SimulationController
import markov.simulation.SimulationTime
import java.awt.TrayIcon
import java.io.InputStream

fun runApplication(mapStream: InputStream, movementStream: InputStream, simulationTime: SimulationTime) {
    try {
        val map = SimulationMapController(Console).readMap(MapReader.read(mapStream))
        val movement =
            MovementController(
                map!!.size,
                Console
            ).readMovement(MovementReader.read(movementStream))
        CostMapController(Console).findCostMap(map, movement!!)
        SimulationController(Console).startFrom(map, SimulationTime(10), movement, OneToHundredRandomGenerator)
    } catch (_: IllegalArgumentException) {
        println("오류가 발생해 프로그램을 종료합니다")
    }
}

fun main(args: Array<String>) {
    if (args.size != 1) {
        println("시뮬레이션 시간만을 인자로 전달해야합니다.")
    }
    runApplication(DataLoader.load(Data.MAP), DataLoader.load(Data.PROBABILITY), SimulationTime.of(args[0]))
}
